--- intel-vaapi-driver-2.3.0-org/src/shaders/post_processing/gen7/RGB_to_YUV.g4a	2019-01-02 11:54:01.782534921 -0800
+++ intel-vaapi-driver-2.3.0/src/shaders/post_processing/gen7/RGB_to_YUV.g4a	2019-01-02 11:57:44.982538364 -0800
@@ -399,20 +399,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -433,20 +434,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -466,20 +468,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -499,20 +502,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -535,20 +539,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -569,20 +574,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -602,20 +608,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -635,20 +642,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -670,20 +678,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -703,20 +712,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -736,20 +746,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -769,20 +780,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -804,20 +816,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -838,20 +851,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -871,20 +885,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -904,20 +919,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
--- intel-vaapi-driver-2.3.0-org/src/shaders/post_processing/gen8/RGB_to_YUV.g8a	2018-12-09 18:17:18.000000000 -0800
+++ intel-vaapi-driver-2.3.0/src/shaders/post_processing/gen8/RGB_to_YUV.g8a	2019-01-02 11:30:15.948065512 -0800
@@ -377,20 +377,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -411,20 +412,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -444,20 +446,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -477,20 +480,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -513,20 +517,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -547,20 +552,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -580,20 +586,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -613,20 +620,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -648,20 +656,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -681,20 +690,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -714,20 +724,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -747,20 +758,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -782,20 +794,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 17]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 1]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 17]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -816,20 +829,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 49]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 33]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 49]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -849,20 +863,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 81]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 65]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 81]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
@@ -882,20 +897,21 @@
         mov   (8) fBUFFER_G(1, 0)<1>	  r[a0.2, 113]<16;8,2>:ub          
         mov   (8) fBUFFER_B(0, 0)<1>	  r[a0.0, 97]<16;8,2>:ub           
         mov   (8) fBUFFER_B(1, 0)<1>	  r[a0.0, 113]<16;8,2>:ub           
-	
-	mul   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.299f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	0.587f
-	mac	(16) fBUFFER_Y(0, 0)<1>	  fBUFFER_B(0, 0)<8;8,1>	0.114f
-	
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	-0.169f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.331f
-	mac   (16) fBUFFER_U(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	0.499f
-
-	mov   (16) acc0.0<1>:f		128.0f	
-	mac   (16) acc0.0<1>:f		fBUFFER_R(0, 0)<8;8,1>	0.499f
-	mac   (16) acc0.0<1>:f		fBUFFER_G(0, 0)<8;8,1>	-0.418f
-	mac   (16) fBUFFER_V(0, 0)<1>	fBUFFER_B(0, 0)<8;8,1>	-0.0813f
+
+        mov   (16) acc0.0<1>:f          16.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.257f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  0.504f
+        mac   (16) fBUFFER_Y(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.098f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  -0.148f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.291f
+        mac   (16) fBUFFER_U(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  0.439f
+
+        mov   (16) acc0.0<1>:f          128.0f
+        mac   (16) acc0.0<1>:f          fBUFFER_R(0, 0)<8;8,1>  0.439f
+        mac   (16) acc0.0<1>:f          fBUFFER_G(0, 0)<8;8,1>  -0.368f
+        mac   (16) fBUFFER_V(0, 0)<1>   fBUFFER_B(0, 0)<8;8,1>  -0.071f
 
 	mov     (16)    wTempY(0,0)<2>		fBUFFER_Y(0, 0)<8;8,1>
 	mov     (16)    wTempU(0,0)<2>		fBUFFER_U(0, 0)<8;8,1>
